### DISCUSSION TITLE
How do I configure dagster itself and my jobs to run in different K8s namespaces

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18220] [Q&A] [answered]
Hey all - have been away from Dagster for a bit working on other things but am back now and revisiting something I tried to get working ages ago but couldn't: running jobs in a separate k8s namespace from where Dagster itself is running.
I'm using the standard helm chart with separate user code repositories. I've set the `jobNamespace` param in the config section for the `K8sRunLauncher` and I've followed the other config changes mentioned here: <https://github.com/dagster-io/dagster/issues/4340> (`dagster` and `dagster-jobs` are the two namespaces). I've got a standalone service account configured with all the correct permissions in both namespaces and set it as the `global` service account in both charts but, still the launcher does not seem to even try to launch in my jobs namespace (should be `dagster-jobs` not `dagster`):

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700618948341699?thread_ts=1700618948.341699&cid=C01U954MEER).*

**Comment by gibsondan** [comment, marked as answer]
It looks like you are running into the issue described here: https://github.com/dagster-io/dagster/issues/13921 By default, the job is prefering to use the namespace specified by the user code deployment, and that is overriding the namespace set on the run launcher. There are some workarounds in the linked issue, including disabling the "includeConfigInLaunchedRuns" flag.
